lib/phoenix/endpoint/supervisor.ex

```diff
@@ -381,25 +390,38 @@ defmodule Phoenix.Endpoint.Supervisor do
   end
 
   defp warmup_static(endpoint) do
-    for {key, value} <- cache_static_manifest(endpoint) do
+    warmup_static(endpoint, cache_static_manifest(endpoint))
+    endpoint.static_path("/")
+  end
+
+  defp warmup_static(endpoint, %{"latest" => latest, "digests" => digests}) do
+    Enum.each(Map.keys(latest), fn key ->
       Phoenix.Config.cache(endpoint, {:__phoenix_static__, "/" <> key}, fn _ ->
-        {:cache, "/" <> value <> "?vsn=d"}
+        {:cache, static_cache(digests, Map.get(latest, key))}
       end)
-    end
-    endpoint.static_path("/")
+    end)
+  end
+
+  defp warmup_static(_endpoint, _manifest) do
+    raise ArgumentError, "expected warmup_static/2 to include 'latest' and 'digests' keys in manifest"
   end
 
+  defp static_cache(digests, value) do
+    {
+      "/#{value}?vsn=d",
+      static_integrity(get_in(digests, [value, "sha512"]))
+    }
+  end
+
+  defp static_integrity(nil), do: ""
```

Comment:
It would be better to return nil in these cases, no?

Reply:
According to the spec, it should either be an empty string or a valid metadata. I'm unsure how eex handles nil values for attributes, so I went with the option I knew would work. _I think it should work_ if given a `nil` but am not positive.

https://www.w3.org/TR/SRI/#the-integrity-attribute

Reply:
It should work with nil, yes. Nil is converted into an empty string by EEx. Even if it wasn't, then users could convert it. It is best for us to return the information in the best format possible instead of trying to foresee how it is used by the view.

Reply:
Ok, then it should clearly return `nil`. Will update.